docs/cli.md

```diff
@@ -29,17 +34,303 @@ Common Flags
+
+```
+$ oc deploy <deployment-config> [options]
+```
+
+If invoked without options, display the latest deployment for the specified *deployment-config*.
+For example:
+
+```
+$ oc deploy database
+```
+
+Flag `--latest` starts a deployment.
+Flag `--retry` retries the latest failed deployment.
+Flag `--cancel` cancels the in-progress deployment.
+The following example shows how to cancel the deployment mentioned in the preceding example:
+
+```
+$ oc deploy database --cancel
+```
+
+### oc rollback
+
+This reverts the pod and container configuration back to a previous deployment.
+Scaling and trigger settings are normally left as-is.
+Flag `--dry-run` displays what the rollback *would do* but does not perform the rollback.
+Flags `--change-triggers` and `--change-strategy` include the previous deployment's triggers and strategies, respectively, in the rollback.
+Flag `--change-scaling-settings` includes the previous deployment's replication controller replica count and selector in the rollback.
+
+If specified, option `--output <format>` inhibits the rollback and instead displays updated deployment configuration in the specified *format*:
+
+* `json`
+* `yaml`
+* `template`
+* `templatefile`
+
+With `--output template`, option `-t <template-string>` specifies a template string.
+With `--output templatefile`, option `-t <filename>` specifies a template file.
+
+Here are two examples:
+
+```
+# Perform a rollback.
+$ oc rollback deployment-1
+
+# Perform the rollback "manually" by piping back to "oc replace".
+$ oc rollback deployment-1 --output=json | oc replace dc deployment -f -
+```
+
+See also [`oc replace`](#oc-replace).
+
+### oc new-build
+
+This creates a new build with the specified source code.
+It creates a build configuration for your application using images and code that has a public repository.
+It looks up the images on the local Docker installation (if available), a Docker registry, or an OpenShift image stream.
+If you specify a source code URL, it sets up a build that takes the source code and converts it into an image that can run inside a pod.
+Local source must be in a git repository that has a remote repository that the OpenShift instance can see.
+
+Once the build configuration is created you may need to run a build with `oc start-build`.
+
+The general form is:
+
+```
+$ oc new-build <component> [options]
+```
+
+where *component* has one of the forms:
+
+| Form            | Description           |
+|:----------------|:----------------------|
+| *image*         | Use *image* directly. |
+| *imagestream*   | Use the latest image in *imagestream*. |
+| *path*          | Use source code found at *path*. |
+| *url*           | Use source code found at *url*. |
+| *image*`~`*url* | Note the tilde `~` between *image* and *url*.  In this case the component is actually made of two sub-components, the *image* and the source code found at *url*.  Use the image as the base and arrange to build the source code on it. |
```

Comment:
Don't mix and match regular text with code fragments, so instead of `*image*'~'*url*` just do `*image~url*`